few improvements

### 1. What version of Caddy are you using (`caddy -version`)?
master

### 2. What are you trying to do?
small improvements

### 3. What is your entire Caddyfile?
 non apliable

### 4. How did you run Caddy (give the full command and describe the execution environment)?
 non apliable

### 5. Please paste any relevant HTTP request(s) here.
 non apliable

### 6. What did you expect to see?
 non apliable

### 7. What did you see instead (give full error messages and/or log)?
 non apliable

### 8. How can someone who is starting from scratch reproduce the bug as minimally as possible?
yes, install [`megacheck`](https://github.com/dominikh/go-tools/tree/master/cmd/megacheck) and run it in master as `megacheck -simple.enabled`

output by `megacheck`:
```
➜  caddy git:(master) megacheck -simple.enabled
caddy.go:569:2: 'if err != nil { return err }; return nil' can be simplified to 'return err' (S1013)
sigtrap_posix.go:87:5: this value of inst is never used (SA4006)
upgrade.go:151:5: should omit nil check; len() for nil slices is defined as zero (S1009)
```

--- a/caddy.go
+++ b/caddy.go
@@ -565,12 +565,7 @@ func ValidateAndExecuteDirectives(cdyfile Input, inst *Instance, justValidate bo
 		return err
 	}
 
-	err = executeDirectives(inst, cdyfile.Path(), stype.Directives(), sblocks, justValidate)
-	if err != nil {
-		return err
-	}
-
-	return nil
+	return executeDirectives(inst, cdyfile.Path(), stype.Directives(), sblocks, justValidate)
 }
 
 func executeDirectives(inst *Instance, filename string,

--- a/sigtrap_posix.go
+++ b/sigtrap_posix.go
@@ -84,7 +84,7 @@ func trapSignalsPosix() {
 				}
 
 				// Kick off the restart; our work is done
-				inst, err = inst.Restart(caddyfileToUse)
+				_, err = inst.Restart(caddyfileToUse)
 				if err != nil {
 					log.Printf("[ERROR] SIGUSR1: %v", err)
 				}

--- a/upgrade.go
+++ b/upgrade.go
@@ -148,7 +148,7 @@ func Upgrade() error {
 
 	// determine whether child startup succeeded
 	answer, readErr := ioutil.ReadAll(sigrpipe)
-	if answer == nil || len(answer) == 0 {
+	if len(answer) == 0 {
 		cmdErr := cmd.Wait() // get exit status
 		errStr := fmt.Sprintf("child failed to initialize: %v", cmdErr)
 		if readErr != nil {

